hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClient.java

```diff
@@ -323,4 +325,19 @@ default void assertValidKeysAndCertificate() throws OzoneSecurityException {
    */
   boolean processCrl(CRLInfo crl);
 
+  /**
+   * Return the store factory for with key manager and trust manager for server.
+   */
+  default KeyStoresFactory getServerKeyStoresFactory()
```

Comment:
I don't think we need the default implementation for these two methods, the only place where we use it is in the CertificateClientTest class, other implementors are either implementing these methods, or inherit it from the DefaultCertificateClient, so I believe CertificateClientTest can have its own implementation that returns null, what do you think?